Tests are failing due to zarr-related(?) module import

Since 2 days ago, our tests are suddenly failing with a module import error, potentially related to zarr:
```
tests/test_kernels_validation.py:6: in <module>
    import zarr
/usr/share/miniconda/envs/gcm-filters-env/lib/pypy3.9/site-packages/zarr/__init__.py:2: in <module>
    from zarr.codecs import *
/usr/share/miniconda/envs/gcm-filters-env/lib/pypy3.9/site-packages/zarr/codecs.py:2: in <module>
    from numcodecs import *
/usr/share/miniconda/envs/gcm-filters-env/lib/pypy3.9/site-packages/numcodecs/__init__.py:100: in <module>
    from numcodecs.shuffle import Shuffle
/usr/share/miniconda/envs/gcm-filters-env/lib/pypy3.9/site-packages/numcodecs/shuffle.py:4: in <module>
    from ._shuffle import _doShuffle, _doUnshuffle
E   ModuleNotFoundError: No module named 'numcodecs._shuffle'
```

Any ideas what is going on @rabernat @jbusecke @paigem @pbranson?

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -9,6 +9,7 @@ flake8-print
 interrogate
 isort
 nbsphinx
+numcodecs==0.10.1
 pre-commit
 pylint
 pytest

